Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/TestsController.cs

```diff
@@ -675,17 +659,28 @@ public ActionResult Import(string problemId, HttpPostedFileBase file, bool retes
                     this.TempData.AddDangerMessage(Resource.Invalid_tests);
                     return this.RedirectToAction("Problem", new { id });
                 }
+            }
+
+            using (var scope = new TransactionScope())
+            {
+                if (deleteOldFiles)
+                {
+                    var testsIds = this.Data.Tests.All().Where(t => t.ProblemId == problem.Id).Select(t => t.Id).ToList();
+                    this.Data.TestRuns.Delete(tr => testsIds.Contains(tr.TestId));
+                    this.Data.Tests.Delete(t => testsIds.Contains(t.Id));
+                }
 
                 ZippedTestsManipulator.AddTestsToProblem(problem, parsedTests);
 
-                this.Data.SaveChanges();
-            }
+                if (retestTask)
+                {
+                    this.RetestSubmissions(problem.Id);
```

Comment:
Reviewed it with @KristianMariyanov and he confirmed that multiple SaveChanges() are ok.